app/worker/utils.go

```diff
@@ -15,17 +15,57 @@
 package worker
 
 import (
-	"fmt"
+	"os"
+	"regexp"
 	"strings"
 
 	"github.com/mendersoftware/workflows/model"
+	"github.com/thedevsaddam/gojsonq"
 )
 
+const (
+	workflowEnvVariable   = "env."
+	workflowInputVariable = "workflow.input."
+)
+
+var reExpression = regexp.MustCompile(`\$\{([^\}]+)\}`)
+var reExpressionOutput = regexp.MustCompile(`(.*)\.output\.(.*)`)
+
 func processJobString(data string, workflow *model.Workflow, job *model.Job) string {
-	for _, param := range job.InputParameters {
-		data = strings.ReplaceAll(data,
-			fmt.Sprintf("${workflow.input.%s}", param.Name),
-			param.Value)
+	matches := reExpression.FindAllStringSubmatch(data, -1)
+
+	for _, submatch := range matches {
+		match := submatch[1]
+		if strings.HasPrefix(match, workflowInputVariable) && len(match) > len(workflowInputVariable) {
+			paramName := match[len(workflowInputVariable):len(match)]
+			for _, param := range job.InputParameters {
+				if param.Name == paramName {
+					data = strings.ReplaceAll(data, submatch[0], param.Value)
+					break
+				}
+			}
+		} else if strings.HasPrefix(match, workflowEnvVariable) && len(match) > len(workflowEnvVariable) {
+			envName := match[len(workflowEnvVariable):len(match)]
+			envValue := os.Getenv(envName)
+			data = strings.ReplaceAll(data, submatch[0], envValue)
+		} else if output := reExpressionOutput.FindStringSubmatch(match); len(output) > 0 {
+			for _, result := range job.Results {
+				if result.Name == output[1] {
+					varKey := output[2]
+					var output string
+					if result.Type == model.TaskTypeHTTP {
+						output = result.HTTPResponse.Body
+					} else if result.Type == model.TaskTypeCLI {
+						output = result.CLI.Output
+					} else {
+						continue
+					}
+					varValue := gojsonq.New().FromString(output).Find(varKey)
```

Comment:
for portability point of view I would check if the Content-Type is application/json otherwise I am not sure how it will behave.